submissions/JOSEF/Alt-F4 - JOSEF.md

```diff
@@ -0,0 +1,156 @@
+
+- In this version, I relied quite heavily on bots transporting things across the whole base (mainly to stock up the builder train and return unused materials from newly analyzed cells). This was more due to my lazyness than anything else and I‘m definitely planning on improving that for future versions. But my system of spamming production cells everywhere, using an excessive amount of construction bots and giving them access to building materials scattered across the whole base worked a lot better than I had expected – Build jobs are given to nearby bots which always take things from the closest location available, as long as you only offer them one type of chest.
+
+- I loved coming up with the system, designing the cells and all that, but I have to be honest: Testing something like this is a bit annoying, especially if you‘re impatient and want to see it running. It‘s like testing a Rube Goldberg machine that runs for many hours – For every tiny change that you make, you have to start over, implement the change and let it run for a couple of hours. Many small fixes you quickly come up with will lead to more unforeseen problems that might only occur in ring 6 or so. You can of course make things easier by e.g. giving it infinite construction materials when you‘re testing the ore finding mechanism, but actual realistic tests are pretty tedious.
+
+This is certainly not the last JOSEF I‘m ever going to build. In fact, I started a small, unprofessional [tutorial series](https://youtube.com/playlist?list=PLqiMv9sOILspVylhnHefjleLWw1Ye2RrD) in which I‘m building a second version. I‘m not yet sure where this will go but I have a few ideas:
+
+- Larger cells. I'm currently planning on doing 4x4 chunks instead of 2x2, but I'm not sure if I'll stick with that. I’ll have uncoupled bot networks to begin with which will greatly simplify the whole ore finding process.
+
+- Dealing with water, both as a resource and as an obstacle: Should be very doable
+
+- Dealing with biters: Will be difficult to get right. I won‘t tackle deathworlds but small amounts of biters should be okay
+
+- Dealing with cliffs: Will only be difficult psychologically. I‘ve always hated them and turned them off right away. Should be easy to handle.
+
+- Making science: Will be a little tricky, simply because it will hugely increase the draw on resources and the amount of active trains. I will have to heavily limit the science generated in individual cells. Also, obviously, production materials have to be prioritized from science or the thing will grind to a halt.
+
+## Final Thoughts
```

Comment:
I really like how you close off your article :) 

But I just want to mention, just because Josef is is simpler - or _less smart_ - than GreyGoo doesn't mean it was less of an achievement. It's a complex problem that Josef had to solve, and if you find the simplest rules that allows that then technically your solution is better than GreyGoo. Einstein says that to truley understand something is to be able to explain it simply, and you can do that. That doesn't take anything from GreyGoo, he came up with a new never-before-seen-idea and it was awesome. And you built on top of his design, simplified it, and created a very robust and capable system from it. Well done :)

I hope we will actually see a Josef Mk2!